### ISSUE TITLE
Desktop: Keyboard navigation is broken

### ISSUE THREAD

**Issue by bsavant** [laurent22/joplin#16407]
### Operating system

macOS

### Joplin version

Joplin 3.7.14

### Desktop version info

Joplin 3.7.14 (prod, darwin)

Device: darwin, Apple M4 Pro
Client ID: 725103b096e443d98d7171d465db3106
Sync Version: 3
Profile Version: 53
Keychain Supported: Yes
Alternative instance ID: -
Sync target: Joplin Server
Editor: Markdown

Revision: 205c284

### Current behaviour

Sample Note contents:
```
# Test 1
# Test 2
# Test 3
# Test 4
# Test 5
# Test 6
- test
- test
- test
- test
```

I can consistently reproduce this with a minimum of 6 Heading 1's in a note.  If you arrow down through the note to the bottom, and then hit the up arrow, it puts the cursor at `# Test 6` line instead of navigating through the list.  When you use Cmd+Left/Right Arrows (Home/End) in the list entries, it drops you to the next line.

Note: All plugins were disabled for this testing.

https://github.com/user-attachments/assets/c2767cee-21ce-4a12-ad83-689d88a1bf81

### Expected behaviour

Keyboard navigation should be consistent.

**Comment by personalizedrefrigerator** [comment]
I can reproduce this locally, both with and without "Markdown Editor: render markup in editor" enabled. For me, this happens with not just the last line, but all lines after the "# Test 6" header.

**Update**: This is also an issue in Joplin v3.7.6.

**Comment by personalizedrefrigerator** [comment]
This seems to be a regression (does not occur in v3.6.16).

**Comment by personalizedrefrigerator** [comment]
Based on a `git bisect`, this is a regression from 698b59b8fc35278546609f8a7a2d87746d4bf272.
